File: Practica_3/java/Cola.java
package Practica_3.java;

public class Cola {
    private long arreglo[];
    private int inicio;
    private int fin;
    private int n;

    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n+1];
        this.inicio = 0;
        this.fin = -1;
    }
    public void insert(long e) {
        if (this.fin==this.n) {
            System.out.println("Cola llena");
        } else {
            this.arreglo[this.fin] = e;
            this.fin++;
        }
    }
    public long remove(){
        if(this.inicio==0 && this.fin ==0)
        {
            System.out.println("Cola vacía");
            return -1;
        }else{
            this.inicio += 1;
            long dato= this.arreglo[this.inicio];
            if(this.inicio==this.fin){
                this.inicio = 0;
                this.fin = 0;
            }return dato;
        }
    }
    public  long peek(){
        return this.arreglo[this.inicio+1];

    }
    public boolean isEmpty(){
        return this.inicio==this.fin;
    }
    public boolean isFull(){
        return this.fin==this.n;
    }
    public int size(){
        return this.fin-this.inicio;
    }
    
}

File: recursoso/Cola.py
package recursoso;

public class Cola:
def __init__(self, n):
    self.__arreglo = [0] * (n + 1)  # todo esto es tu constructor
    self.__inicio = 0
    self.__fin = 0
    self.__n = n

def insert(self, e):
    if self.__fin == self.__n:
        print("Cola llena")
    else:
        self.__fin += 1
        self.__arreglo[self.__fin] = e

def remove(self):  # consultar datos a sus atributos
    if self.__inicio == 0 and self.__fin == 0:
        print("Cola vacía")
        return None
    else:
        self.__inicio += 1
        dato = self.__arreglo[self.__inicio]
        if self.__inicio == self.__fin:
            self.__inicio = 0
            self.__fin = 0
        return dato

def peek(self):  # devuelve el dato que esté hasta el frente de la cola
    return self.__arreglo[self.__inicio + 1]

def isEmpty(self):  # devuelve verdadero si la cola está vacía
    return self.__inicio == self.__fin == 0

def isFull(self):  # cola llena verifica
    return self.__fin == self.__n

def size(self):  # cuántos elementos tiene
    return self.__fin - self.__inicio

{
    
}
